src/AWS/CloudWatch/Cloudwatch.purs

```diff
@@ -1,44 +1,38 @@
-module AWS.Cloudwatch where
+module AWS.CloudWatch where
 
 import Prelude
-import AWS.Core (AccessKeyId(..), InstanceId, Region(..), SecretAccessKey(..), SessionToken(..))
+
+import AWS.Core.Client (makeClientHelper, makeDefaultClient)
+import AWS.Core.Types (DefaultClientProps, DefaultClientPropsR, InstanceId)
 import Control.Promise (Promise)
 import Control.Promise as Promise
-import Data.Bifunctor (bimap)
+import Data.Bifunctor (lmap)
 import Data.DateTime (DateTime)
 import Data.Either (Either)
 import Data.Foldable (fold)
-import Data.Function.Uncurried (Fn1, Fn2, runFn2)
+import Data.Function.Uncurried (Fn2, runFn2)
 import Data.JSDate (JSDate, fromDateTime)
 import Data.List.Types (NonEmptyList)
-import Data.Maybe (Maybe)
-import Data.Newtype (un, unwrap)
-import Data.Nullable (Nullable, toNullable)
+import Data.Newtype (unwrap)
 import Effect (Effect)
 import Effect.Aff (Aff)
 import Effect.Class (liftEffect)
-import Foreign (ForeignError)
+import Foreign (Foreign, ForeignError)
+import Justifill.Fillable (class FillableFields)
+import Justifill.Justifiable (class JustifiableFields)
+import Prim.Row (class Union)
+import Prim.RowList (class RowToList)
 import Simple.JSON (readJSON)
 
-foreign import data Cloudwatch :: Type
-
-type InternalMakeClientParams
-  = { region :: String
-    , secretAccessKey :: String
-    , accessKeyId :: String
-    , sessionToken :: Nullable String
-    }
+foreign import data CloudWatch :: Type
 
-foreign import makeClientImpl :: Fn1 InternalMakeClientParams (Effect Cloudwatch)
+foreign import newCloudWatch :: Foreign -> (Effect CloudWatch)
 
-makeClient :: Region -> AccessKeyId -> SecretAccessKey -> Maybe SessionToken -> Effect Cloudwatch
-makeClient r a s t =
-  makeClientImpl
-    { region: (un Region r)
-    , secretAccessKey: (un SecretAccessKey s)
-    , accessKeyId: (un AccessKeyId a)
-    , sessionToken: toNullable $ map (un SessionToken) t
-    }
+makeClient :: forall t4 t5 t6 t7 t8.
```

Comment:
Can we name the type params here beginning from 0? :thinking: 4 feels odd :smile_cat: